extensions/schema-compare/src/main.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+import * as vscode from 'vscode';
+import MainController from './controllers/mainController';
+
+let controllers: MainController[] = [];
+
+export function activate(context: vscode.ExtensionContext) {
+	let activations: Promise<boolean>[] = [];
+
+	// Start the main controller
+	let mainController = new MainController(context);
+	controllers.push(mainController);
+	context.subscriptions.push(mainController);
+	activations.push(mainController.activate());
```

Comment:
>activations.push(mainController.activate()); [](start = 1, length = 44)

You only have this one activation - suggest getting rid of all this Promise stuff and just await on the activate call directly